yarn.config.cjs

```diff
@@ -142,6 +144,38 @@ function enforceWorkspaceDependencies({Yarn}) {
       }
     }
   }
+
+  // This finds all the peer dependencies and checks that everything up the
+  // workspace tree has the same peer dependency.
```

Comment:
Doesn't yarn itself check these? How did you find them in the first place?

Reply:
I thought yarn should catch these, but for some reason it didn't, I don't know why.

Found them through the issue that was opened against us and by running the command I listed in the description on their reproduction repo.

I'd like to get this fixed for the user and then spend some time figuring out if there is a bug in yarn, possibly upgrading to a newer version would fix it.